domain-management/src/main/java/org/jboss/as/domain/management/security/adduser/JavaConsole.java

```diff
@@ -59,26 +61,34 @@ public void printf(String format, Object... args) throws IllegalFormatException
 
     @Override
     public String readLine(String fmt, Object... args) throws IOError {
-        if (hasConsole()) {
-            return theConsole.readLine(fmt, args);
-        } else {
-            throw ROOT_LOGGER.noConsoleAvailable();
+        ReadLineHandler readLineHandler = new ReadLineHandler(new Prompt(String.format(fmt, args)));
+        return readInputLine(readLineHandler);
+    }
+
+    private String readInputLine(ReadLineHandler readLineHandler) {
+        try {
+            createTerminalConnection(readLineHandler);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
this should log a proper error - you can find an example on the line 65 before change